pkg/server/plugin/upstreamca/awssecret/awslib_fake.go

```diff
@@ -0,0 +1,61 @@
+package awssecret
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+	"github.com/stretchr/testify/suite"
+)
+
+type fakeSecretsManagerClient struct {
+	suite.Suite
+	secretsmanageriface.SecretsManagerAPI
```

Comment:
instead of embedding this interface, it would be cleaner to simply define our own interface with the single method needed to do the job. The function that creates the new client could return that interface instead, making it much more clear what our dependency is.